Implementations for Global Sitecore.Foundation.SitecoreExtensions project into Headstart Api solution.

ppatel-sitecore opened the pull request:
1. Completed: Added the implementation of the new 'Sitecore.Foundation.SitecoreExtensions' project to the 'Headstart' solution, in order to leverage the global logging mechanism. This includes Resharper code-analysis-based refactoring for optimized coding practices and standards for the 'Sitecore.Foundation.SitecoreExtensions' project.
2. Completed Code clean-up for all classes for the 'Headstart.API' project to correctly apply coding formatting and standards (i.e. removing unneeded code comments, adding missing code comments, and removing un-used namespace references). This includes Resharper code-analysis-based refactoring for optimized coding practices and standards for the 'Headstart.API' project.
3. Completed: Code clean-up for all interfaces/implementations classes for the 'Headstart.API' project to correctly apply coding formatting and standards (i.e. removing unneeded code comments, adding missing code comments, and removing un-used namespace references).
4. Completed: Adding the implementation of the logging mechanism (try/catch logging) from the 'Sitecore.Foundation.SitecoreExtensions' project to all the classes for the 'Headstart.API' and 'Headstart. Common' projects.
5. Completed: Code clean-up for all classes for the 'Headstart. Common' project to correctly apply coding formatting and standards (i.e. removing any unneeded code comments and removing un-used namespace references). This includes Resharper code-analysis-based refactoring for optimized coding practices and standards for the 'Headstart. Common' project.

Commit:
1. Add the implementation of the new 'Sitecore.Foundation.SitecoreExtensions' project to the 'Headstart' solution, in order to leverage the global logging mechanism.
2. Update all controller classes for the 'Headstart.API' project to correctly apply coding formatting and standards (i.e. code comments and removing un-used namespace references).

Commit:
1. Added a new IsNonProdEnv appSettings configurations.

Commit:
1. Added the implementation of the new 'Sitecore.Foundation.SitecoreExtensions' project to the 'Headstart' solution, in order to leverage the global logging mechanism.
2. Updated all controller classes for the 'Headstart.API' project to correctly apply coding formatting and standards (i.e. code comments and removing un-used namespace references).
3. Updated all interfaces/implementations classes for the 'Headstart.API' project to correctly apply coding formatting and standards (i.e. code comments and removing un-used namespace references).
4. Added the implementation of the logging mechanism (try/catch logging) from the 'Sitecore.Foundation.SitecoreExtensions' project, to all the classes for the 'Headstart.API' project.

Commit:
1. Code clean-up for all interfaces/implementations classes for the 'Headstart.API' project to correctly apply coding formatting and standards (i.e. removing unneeded code comments, adding missing code comments and removing un-used namespace references).
2. Code clean-up for all classes for the 'Headstart.Common' project to correctly apply coding formatting and standards (i.e. removing unneeded code comments and removing un-used namespace references).

Commit:
1. Updated the implementation of the new 'Sitecore.Foundation.SitecoreExtensions' project to the 'Headstart' solution, in order to leverage the global logging mechanism. This includes Resharper code-analysis based recaforing for optimized coding practices and standards for the 'Sitecore.Foundation.SitecoreExtensions' project.
2. WIP: Code clean-up for all icontroller classes for the 'Headstart.API' project to correctly apply coding formatting and standards (i.e.  removing unneeded code comments, adding missing code comments and removing un-used namespace references). This includes Resharper code-analysis based recaforing for optimized coding practices and standards for the 'Headstart.API' project.
3. WIP: Code clean-up for all interfaces/implementations classes for the 'Headstart.API' project to correctly apply coding formatting and standards (i.e. removing unneeded code comments, adding missing code comments and removing un-used namespace references).
4. WIP: Code clean-up for all classes for the 'Headstart.Common' project to correctly apply coding formatting and standards (i.e. removing unneeded code comments and removing un-used namespace references). This includes Resharper code-analysis based recaforing for optimized coding practices and standards for the 'Headstart.API' project.

Commit:
1. WIP: Code clean-up for all icontroller classes for the 'Headstart.API' project to correctly apply coding formatting and standards (i.e.  removing unneeded code comments, adding missing code comments and removing un-used namespace references). This includes Resharper code-analysis based recaforing for optimized coding practices and standards for the 'Headstart.API' project.

Commit:
1. Completed: Added the implementation of the new 'Sitecore.Foundation.SitecoreExtensions' project to the 'Headstart' solution, in order to leverage the global logging mechanism. This includes Resharper code-analysis based recaforing for optimized coding practices and standards for the 'Sitecore.Foundation.SitecoreExtensions' project.
2. Completed Code clean-up for all icontroller classes for the 'Headstart.API' project to correctly apply coding formatting and standards (i.e.  removing unneeded code comments, adding missing code comments and removing un-used namespace references). This includes Resharper code-analysis based recaforing for optimized coding practices and standards for the 'Headstart.API' project.
3. Completed: Code clean-up for all interfaces/implementations classes for the 'Headstart.API' project to correctly apply coding formatting and standards (i.e. removing unneeded code comments, adding missing code comments and removing un-used namespace references).
4. Completed: Adding the implementation of the logging mechanism (try/catch logging) from the 'Sitecore.Foundation.SitecoreExtensions' project, to all the classes for the 'Headstart.API' and 'Headstart.Common' projects.
5. Completed: Code clean-up for all classes for the 'Headstart.Common' project to correctly apply coding formatting and standards (i.e. removing unneeded code comments and removing un-used namespace references). This includes Resharper code-analysis based recaforing for optimized coding practices and standards for the 'Headstart.Common' project.

Commit:
1. Completed Code clean-up for all classes for the 'Headstart.API' project to correctly apply coding formatting and standards (i.e. removing unneeded code comments, adding missing code comments, and removing un-used namespace references). This includes Resharper code-analysis-based refactoring for optimized coding practices and standards for the 'Headstart.API' project.

Commit:
1. Completed: Added the implementation of the new 'Sitecore.Foundation.SitecoreExtensions' project to the 'Headstart' solution, in order to leverage the global logging mechanism. This includes Resharper code-analysis-based refactoring for optimized coding practices and standards for the 'Sitecore.Foundation.SitecoreExtensions' project.

Commit:
1. Completed: Added the implementation of the new 'Sitecore.Foundation.SitecoreExtensions' project to the 'Headstart' solution, in order to leverage the global logging mechanism. This includes Resharper code-analysis-based refactoring for optimized coding practices and standards for the 'Sitecore.Foundation.SitecoreExtensions' project.

Commit:
1. Completed: Added the implementation of the new 'Sitecore.Foundation.SitecoreExtensions' project to the 'Headstart' solution, in order to leverage the global logging mechanism. This includes Resharper code-analysis-based refactoring for optimized coding practices and standards for the 'Sitecore.Foundation.SitecoreExtensions' project.

Commit:
Removed un-used files

crhistianramirez commented:
@ppatel-sitecore Thanks for your contribution, appreciate the amount of effort and detail you took to make so many changes. 

My one piece of criticism would be to the try/catch logging. It seems to be adding a lot of "noise" to the project. My preference for managing a cross cutting concern such as logging would be to add something at the middleware level instead. I was actually eying [this package](https://github.com/matthiasguentert/ApplicationInsightsRequestLogging) because it does just that and additionally logs it to application insights which is great for analyzing and digging into. I'm noticing the logging you've added is a Sitecore package. If you see some benefits of using that instead, then we could incorporate a similar strategy with that package. What do you think?

By the way it looks like the build is failing because it can't find the Sitecore packages.

Commit:
Fixing nuget source issues for clean clone and automation build

1. Fixing nuget source issues for clean clone and automation build.

Commit:
Fixing NuGet source issues for clean clone and automation build.

1. Fixing NuGet source issues for clean clone and automation build.

ppatel-sitecore committed:
Update azure-pipelines.yml for Azure Pipelines

1. Fixing NuGet source issues for clean clone and automation build.

Commit:
Merge remote-tracking branch 'ordercloud-api/azure-pipelines' into development

Commit:
Fixing NuGet source issues for clean clone and automation build.

1. Fixing NuGet source issues for clean clone and automation build.
2. Completed Code clean-up for all classes for the 'Headstart.Jobs' project to correctly apply coding formatting and standards (i.e. removing unneeded code comments, adding missing code comments, and removing un-used namespace references). This includes Resharper code-analysis-based refactoring for optimized coding practices and standards for the 'Headstart.Jobs' project.

ppatel-sitecore committed:
Merge branch 'development' into development

Commit:
Added fixes for restoring dependent  Sitecore NuGet packages

1. Added fixes for restoring dependent  Sitecore NuGet packages

Commit:
Merge remote-tracking branch 'Headstart-ppatel/development' into development

Commit:
Added fixes for restoring dependent Sitecore NuGet packages

1. Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
Added fixes for restoring dependent Sitecore NuGet packages.

1. Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
Added fixes for restoring dependent Sitecore NuGet packages.

1. Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
WIP: Added fixes for restoring dependent Sitecore NuGet packages.

1. Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
WIP: Added fixes for restoring dependent Sitecore NuGet packages.

1. Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
WIP: Added fixes for restoring dependent Sitecore NuGet packages.

1. Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
WIP: Added fixes for restoring dependent Sitecore NuGet packages.

WIP: Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
WIP: Added fixes for restoring dependent Sitecore NuGet packages.

WIP: Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
Added fixes for restoring dependent Sitecore NuGet packages.

1. Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
1. Added fixes for restoring dependent Sitecore NuGet packages.

1. Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
1. Added fixes for restoring dependent Sitecore NuGet packages.

1. Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
Added fixes for restoring dependent Sitecore NuGet packages.

Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
Added fixes for restoring dependent Sitecore NuGet packages.

Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
Added fixes for restoring dependent Sitecore NuGet packages.

1. Final: Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
1. Final: Added fixes for restoring dependent Sitecore NuGet packages.

1. Final: Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
1. Final: Added fixes for restoring dependent Sitecore NuGet packages.

1. Final: Added fixes for restoring dependent Sitecore NuGet packages.

Commit:
Fixing the headstart-sitecore-foundation-automation-deploy.ps1

Fixing the headstart-sitecore-foundation-automation-deploy.ps1
